### ISSUE TITLE
Provide pre-built Macbook M1 binaries

### ISSUE THREAD

**Issue by mmomtchev** [mmomtchev/node-gdal-async#30]
Waiting on https://github.com/actions/virtual-environments/issues/2187

**Comment by mmomtchev** [comment]
I am looking for a company willing to cover the costs of renting a remote Apple M1 host in order to provide support for the new Apple Macbooks. If paying by the hour, you should expect about 1 or 2 days of uptime per month.

**Comment by alexcroox** [comment]
If anyone is looking for a workaround to get this running locally on M1 Macs:

```
brew install gdal
npm install gdal-async --build-from-source --shared_gdal
```

**Comment by jdesboeufs** [comment]
If you are still interested I can cover these costs with my company.
A Mac Mini M1 (by Scaleway) may do the job?

**Comment by mmomtchev** [comment]
Yes, a Mac Mini M1 by Scaleway would be perfect, I would need a day or two to make it work and certainly less than a day per month for further testing and development.

**Comment by jdesboeufs** [comment]
Sadly they are currently out of stock. I will keep you informed.

<img width="843" alt="Capture d’écran 2022-12-12 à 21 19 56" src="https://user-images.githubusercontent.com/1231232/207146278-bbc5d3dd-ef06-45e2-81d3-0bda1ce6d2c1.png">

**Comment by Radik24** [comment]
Hi,

any progress in M1 build?

Thanks!

**Comment by jdesboeufs** [comment]
Sadly Scaleway doesn't provide M1 instances anymore :(

**Comment by utix** [comment]
Can any ARM VM do the job ?

**Comment by thom4parisot** [comment]
Sounds like it's on [GitHub roadmap for ~~Q3~~ Q4](https://github.com/github/roadmap/issues/528). There used to be a [beta access](https://github.blog/2023-03-01-github-actions-introducing-faster-github-hosted-x64-macos-runners/) but the link is not working anymore.

> Can any ARM VM do the job ?

It should, as long as it can be registered as a GitHub Action self-hosted runner.

**Comment by jdesboeufs** [comment]
<img width="845" alt="Capture d’écran 2023-09-29 à 11 49 38" src="https://github.com/mmomtchev/node-gdal-async/assets/1231232/f1eaa5d8-3a0b-4776-9ec3-39381844ce2a">

@thom4parisot Not in the dashboard.

**Comment by alexcroox** [comment]
https://github.com/actions/runner-images/issues/8439

GitHub action M1 support is now available

**Comment by mmomtchev** [comment]
Github actions support has been available for some time now, but porting without being able to build locally is not really feasible

**Comment by mmomtchev** [comment]
I would just like to remind everyone that I work for free, I have a $2.50 budget per meal and that the French police is intimidating me while all of the users of this project are showing me with random messages that they fully back this extortion.

If anyone wants M1 binaries, you will have to pay for a M1 host.

**Comment by thom4parisot** [comment]
And are you okay with receiving a physical Mac Mini? I suppose it can be setup as a GitHub runner.

I'm happy to contribute to buying/delivering it. If anybody want to tag along 👍🏻

**Comment by mmomtchev** [comment]
Of course that I am, I will support M1 if I can have a build host.

**Comment by zy6p** [comment]
Adrressed as https://github.com/actions/runner-images/issues/8439 and M1 is now available!

**Comment by mmomtchev** [comment]
I still need a build host that I can access to implement the support.

**Comment by mmomtchev** [comment]
It is a pity that most of my users are cheeky criminals who decided to slide with somebody who was extorting me for money (and a particularly distasteful to mention here problem) with the help of the French judiciary and the police against someone who was making them free software. If you want M1 binaries, you will have to find $5/month. Period.

**Comment by lautarodapin-magoya** [comment]
> If anyone is looking for a workaround to get this running locally on M1 Macs:
> 
> ```
> brew install gdal
> npm install gdal-async --build-from-source --shared_gdal
> ```

how do i use this with `pnpm`

**Comment by kgantchev** [comment]
> Adrressed as [actions/runner-images#8439](https://github.com/actions/runner-images/issues/8439) and M1 is now available!

Might I make a suggestion here? You can try the [FlyCI](https://www.flyci.net/)'s M1 and M2 runners which are on average about 2x faster and 2x cheaper than GitHub's runners .  **_We also offer a free tier of 500 mins/month on the M1 runner for OSS projects._**  

## Pricing  

| Processor | vCPU | RAM (GB) | Storage | Label                          | Price on FlyCI | Price on GitHub |
|-----------|------|----------|---------|--------------------------------|----------------|-----------------|
| M1        | 4    | 7        | 28 GB   | flyci-macos-large-latest-m1    | $0.06          | -             |
| M1        | 8    | 14       | 28 GB   | flyci-macos-xlarge-latest-m1   | $0.12          | $0.16           |
| M2        | 4    | 7        | 28 GB   | flyci-macos-large-latest-m2    | $0.08          | -             |
| M2        | 8    | 14       | 28 GB   | flyci-macos-xlarge-latest-m2   | $0.16          | -             |

## Guide for Implementation

1. Install the [FlyCI GitHub app](https://github.com/apps/flyci-prod).
2. Modify the `runs-on` directive in workflow files as demonstrated in this PR.
3. Enjoy 2x faster and 2x cheaper MacOS workflow runs!  

```diff
jobs:
 ci:
-    runs-on: macos-latest
+    runs-on: flyci-macos-large-latest-m1
   steps:
   - name: 👀 Checkout repo
     uses: actions/checkout@v4
```
 
We'll be very happy to have you on board.

**Comment by mmomtchev** [comment]
As you may have seen from my profile, I am currently living on social welfare because of an extortion involving the French police and judiciary about a series of false criminal proceedings - in order to intimidate me people are posting simultaneously comments and issues on my projects - your is simultaneous with a PR from snyk which seem to have accepted to back the intimidation.

**Comment by alexcroox** [comment]
> It is a pity that most of my users are cheeky criminals who decided to slide with somebody who was extorting me for money (and a particularly distasteful to mention here problem) with the help of the French judiciary and the police against someone who was making them free software. If you want M1 binaries, you will have to find $5/month. Period.

@mmomtchev are you still looking for $5 sponsor on github to help provide M1 binaries? I'm happy to set this up as building from source is becoming problematic between machines and gdal updates. Thanks!

**Comment by alexcroox** [comment]
Do you require a dedicated machine (virtual or otherwise) to figure out the build, or are hosted build runners as suggested by others above enough? Just looking for clarification as you mentioned $5 a month above but that doesn't seem enough for a dedicated host that isn't just a build runner?

**Comment by mmomtchev** [comment]
I do not need a permanent build host, Scaleway had an offer with a price per hour.

In fact, I would probably need a day or two in the beginning, then I would be using for a few hours per month.

**Comment by alexcroox** [comment]
I can sponsor $50 per month for a few months to get you up and running before you move to the free tier build servers if that helps? Your GitHub sponsor page has a min $300 amount a month though FYI.

**Comment by mmomtchev** [comment]
No, if you post simultaneously with other people I cannot accept this.

I will explain you my decision making process, which is in fact rather simple:
![](https://i.imgur.com/QONrssS.png)

For example:
* I write open source software for free for people to use it, this allows me raise awareness of the extortion in which the French and EU justice systems are involved -> **YES**
* I accept donations which allow me to reach an ever larger public -> **YES**
* I accept donations from someone who made it clear that he is involved in that extortion -> **NO**

**Comment by mmomtchev** [comment]
I have been through absolutely horrible harassment and I have spent years living on social welfare in order to achieve victory in this affair. Nothing will stop me.

**Comment by alexcroox** [comment]
You sound like you’ve been through hell and are still fighting your own demons, but I don’t know you, we are just here to help push forward your great open source work. The conversations here have nothing to do with your personal life.

**Comment by mmomtchev** [comment]
Well, let me explain you a fundamental principle in criminal law called *"proof beyond any reasonable doubt"*. Many people believe that DNA is an absolute proof. It is not - in fact there is a very small chance that two people can fully share the same DNA - without even being related. The chance is very small - it is on the scale of 1 to trillions. The subject has been studied extensively: [The Evaluation of Forensic DNA Evidence](https://www.ncbi.nlm.nih.gov/books/NBK232610/) - so that in these cases, the explanation that someone just happened to have the same DNA as the real killer does not stand in court.

In your case, barring an eventual activation of the Infinite Improbability Drive, this is absolutely the case.

In fact, since the police tried threatening me about 8 months ago, ***there hasn't been a single comment on any project I own that is not related to this affair***

**Comment by Alek-S** [comment]
> Well, let me explain you a fundamental principle in criminal law called _"proof beyond any reasonable doubt"_. Many people believe that DNA is an absolute proof. It is not - in fact there is a very small chance that two people can fully share the same DNA - without even being related. The chance is very small - it is on the scale of 1 to trillions. The subject has been studied extensively: [The Evaluation of Forensic DNA Evidence](https://www.ncbi.nlm.nih.gov/books/NBK232610/) - so that in these cases, the explanation that someone just happened to have the same DNA as the real killer does not stand in court.
> 
> In your case, barring an eventual activation of the Infinite Improbability Drive, this is absolutely the case.
> 
> In fact, since the police tried threatening me about 8 months ago, _**there hasn't been a single comment on any project I own that is not related to this affair**_

Not to be rude but what does this have to do with getting an arm build?

You say you need money so you can run CI builds, he offers you the money to do it, and you then change your mind and won't take the money and give a kind of crazy-sounding reply on some conspiracy theory that's unrelated.

**Comment by mmomtchev** [comment]
The problem is that I have been living in total isolation for the last 4 years since my ex-employers made a deal with a number of lawyers to drop charges on a series of very explosive lawsuits, and they have been extorting me to shut up about this affair. In order to intimidate me, people are posting simultaneously on my projects. The French police, already heavily involved in this affair, came illegally in my home 8 months ago to intimidate me about this affair. Since this "incident", I have never had even one comment whose goal was something else than this affair. Yours included.

**Comment by mmomtchev** [comment]
This is possible because there was already a deal with the GIS community and the Linux Foundation to try make it look as they did not notice me in order to help the extortion.

**Comment by mmomtchev** [comment]
I think that if someone really needs a macOS ARM64 version, he should be capable of offering to donate an access to a build host without having to post simultaneously to indicate that he is involved in the extortion. In any other case, the only thing that I can say is that making noise about this ongoing extortion is far more important to me than having this macOS ARM64 version.

**Comment by alexcroox** [comment]
@mmomtchev the offer for my organisation (a registered UK charity) to sponser you $50 a month for a few months is still on the table as we are still struggling with getting this working with Apple silicon macs and would really appreciate any focus you can put in this area. Thank you.

**Comment by alexcroox** [comment]
I'm not looking for any recognition of the sponsor, I just want to help you get Apple Silicon builds out the door. If renting the M1 box is barrier, I'm here to remove it.

**Comment by mmomtchev** [comment]
I am getting simultaneous comments with yours. I have lost 5 years of my life because of this extortion and the object of the extortion was the payment of damages for the previous 7 years that I had lost.

It is obviously a difficult decision, but generally politicians, educational institutions and churches have a policy of refusing dirty money when their origin is widely known - without being too strict in their verification process. Usually, they give the money back once it becomes public knowledge. In this particular case, since I am explicitly notified of the nature of the transaction, the right decision would be to refuse.

**Comment by alexcroox** [comment]
Thank you for adding Apple ARM support in the latest version ❤️

**Comment by alexcroox** [comment]
Would you mind publishing v3.10.1 to npm when you get a chance? Thanks!
